Utility/UserInputHandler.cs

```diff
@@ -0,0 +1,37 @@
+﻿// can add (optional) error messages and further input (range) validation
+public static class UserInputHandler
+{
+
+    public static decimal GetValidDecimal(string output)
+    {
+        decimal? input = null;
+        while (!input.HasValue)
+        {
+            Console.Write(output);
+            input = decimal.TryParse(Console.ReadLine(), out decimal d) ? d : new decimal?();
+        }
+        return (decimal)input;
```

Comment:
is this casting needed?

Reply:
casting decimal? to decimal, otherwise, it gives a compilation error